File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

//initial call of function
int get_positive_int(void);

int main(void)
{
    //variable declaration
    int i = get_positive_int();
    int k = i;
    int p = i;

    //line loop
    while (k >= 1)
    {
        //first loop until i characters at k line
        for (int j = 1; j <= i; j++)
        {
            if (j < k)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf(" ");
        printf(" ");
        //second loop until i characters at k line
        while (p >= 1)
        {
            if (p >= k)
            {
                printf("#");
            }
            else
            {
                printf("");
            }
            p = p - 1;
        }
        printf("\n");
        p = i;
        k = k - 1;
    }
}

//check for accepting only positive integer
int get_positive_int(void)
{
    int n;
    //check loop for inputing numbers only between 1 to 8
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    return n;
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    # defining variables 
    h = get_height()
    k = h
    p = h
    
    # line loop
    while k >= 1:
        
        # first loop until h characters at k line
        for j in range(h):
            if j + 1 < k:
                print(" ", end="")
            else:
                print("#", end="")
        print(" ", end="")
        print(" ", end="")
        
        # second loop until h characters at k line
        while p >= 1:
            if p >= k:
                print("#", end="")
            else:
                print("", end="")
            p = p - 1
        print()
        p = h
        k -= 1        
        
# check for valid integer function


def get_height():
    while True:
        n = get_int("Height: ")
        if n > 0 and n < 9:
            break
    return n


main()